src/libopensc/card-iasecc.c

```diff
@@ -1247,8 +1248,13 @@ iasecc_process_fci(struct sc_card *card, struct sc_file *file,
 
 		if (!(mask & acls[0]))
 			continue;
+		/* avoid any access to acls[offs] beyond the taglen */
+		if (offs >= taglen) {
```

Comment:
sorry, but you need to check this *before* accessing the first byte of acls in line 1248!